Fix error dialogue persisting between screens

Treefire33 opened the pull request:
## About The Pull Request
Fixes the error dialogue from persisting between screens.

## Linked Issues
fixes: #3040 

## Proof of Testing
https://github.com/user-attachments/assets/2fbfd244-e153-4aaa-bb26-fb85d318c1ce

### Other
If this isn't the best UX-wise, I do have another solution in mind.

Treefire33 committed:
prevent weird error box ui thing

scribblecrumb commented:
A useful fix, however it would be better to just disable other buttons on the screen while the window is open.

Treefire33 committed:
disable button functionality instead

Instead of killing the UI elements for the error box upon pressing a button, disable the functionality of the other buttons instead.

archanyhm commented:
Also targets the wrong branch, since the bug seems to affect release as well.

scribblecrumb commented:
Changes look good, but it seems that switching the branch pulled a bunch of dev changes over too (transing and other clan save data, specifically)
